react-cors-spa/src/AddProduct.jsx

```diff
@@ -0,0 +1,145 @@
+import DialogContent from '@mui/material/DialogContent';
+import DialogContentText from '@mui/material/DialogContentText';
+import DialogTitle from '@mui/material/DialogTitle';
+import TextField from '@mui/material/TextField';
+import MenuItem from '@mui/material/MenuItem';
+import Select from '@mui/material/Select';
+import FormControl from '@mui/material/FormControl';
+import Box from '@mui/material/Box';
+
+import "./AddProductButton.css";
+import "./ProductDialogue.css";
+import "./AddProduct.css";
+
+import axios from 'axios';
+import { useQuery } from '@tanstack/react-query';
+
+import { useForm } from 'react-hook-form';
+
+function fetchCategories() {
+    return axios.get('http://127.0.0.1:3000/categories');
+}
+
+function AddProduct() {
+    // Handle open/close of dialogue with useState
+    const [open, setOpen] = React.useState(false);
+    const handleClickOpen = () => {
+      setOpen(true);
+    };
+    const handleClose = () => {
+      setOpen(false);
+    };
+
+    // Handle category selection in dialogue with useState
+    const [category, setCategory] = React.useState('');
+    const handleChange = (event) => {
+        setCategory(event.target.value);
+    }
+    
+    // Populate categories from DB with useQuery
+    const { data } = useQuery({
+        queryKey: ['categoryData'],
+        queryFn: fetchCategories,
+    });
+
+    // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort
+    // Scroll to half-way down the page
+    const sortedCategories = data?.data?.sort((a, b) => {
+        const nameA = a.name.toUpperCase(); // ignore upper and lowercase
+        const nameB = b.name.toUpperCase(); // ignore upper and lowercase
+        if (nameA < nameB) {
+          return -1;
+        }
+        if (nameA > nameB) {
+          return 1;
+        }
+      
+        // names must be equal
+        return 0;
+      });
+
+    // Verify form completion with formState
+    const { register, handleSubmit, watch, formState: {errors} } = useForm({
+        defaultValues: {
+            Name: "",
+            SKU: "",
+            Size: "",
+            Description: "",
+            Category: "Category",
+        }
+    });
+
+    console.log(watch());
+
+    // When page is refreshed, initially take 3 seconds to load addProduct button
+    const [loading, setLoading] = React.useState(true);
+
+    React.useEffect(() => {
+        setTimeout(() => {
+            setLoading(false);
+        }, 3000);
+    }, []);
+
+    if(loading) {
+        return (
+            <div className="addProductSkeleton">
+                <Skeleton variant="rounded" width={160} height={75} />
+            </div>
+        )
+    }
+    return (
+        <React.Fragment>
+            <div className="AddProductButtonContainer">
+                <Button variant="outlined" onClick={handleClickOpen} size="large">Add Product</Button>
+            </div>
+            <Dialog open={open}>
```

Comment:
I would probably separate this dialog and its wrapper with the button into different components as they have slight different purposes. Since they are so tightly linked, you could potentially put the both in the same file, but it would also be acceptable to separate them. Up to you.